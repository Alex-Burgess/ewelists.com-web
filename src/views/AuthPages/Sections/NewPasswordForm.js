import React, { useState } from 'react';
import { Auth } from "aws-amplify";
// libs
import { onAuthError } from "libs/errorLib";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// components
import Button from "components/Buttons/Button.js";
import Input from "components/Input/CustomInput.js";
import ErrorText from "components/Typography/Error.js";

import styles from "assets/jss/material-kit-pro-react/views/authPageStyle.js";
const useStyles = makeStyles(styles);

export default function NewPasswordForm(props) {
  const classes = useStyles();
  const { email, setConfirmed } = props;

  const [code, setCode] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [confirmationError, setConfirmationError] = useState('');

  const handleConfirmClick = async event => {
    event.preventDefault();

    if (!(/(?=^[0-9]{6}$)/.test(code))) {
      setConfirmationError("Confirmation code should be a 6 digit number which was sent to you by email.");
      return false
    }

    let passwordError = checkPassword(password);
    if (passwordError) {
      setConfirmationError(passwordError);
      return false
    }

    if (password !== confirmPassword) {
      setConfirmationError("Your confirmed password does not match the new password.");
      return false
    }

    try {
      await Auth.forgotPasswordSubmit(
        email,
        code,
        password
      );

      setConfirmed(true);
    } catch (e) {
      onAuthError(e, email);
      setConfirmationError(e.message);
    }
  };

  const checkPassword = (password) => {
    if (!(/(?=.*[a-z])/.test(password))) {
      return "Password does not contain any lower case letters."
    }

    if (!(/(?=.*[A-Z])/.test(password))) {
      return "Password does not contain any upper case letters."
    }

    if (!(/(?=.*\d)/.test(password))) {
      return "Password does not contain any numbers."
    }

    if (!(/(?=.*[-+_!@#$%^&*.,?])/.test(password))) {
      return "Password does not contain any symbols."
    }

    return null
  }

  const validateResetForm = () => {
    return (
      code.length > 0 &&
      password.length > 6 &&
      confirmPassword.length > 6
    );
  }

  return (
    <form onSubmit={handleConfirmClick}>
      <h3 className={classes.title + " " + classes.textCenter}>Confirmation Code</h3>
      <div id="confirmation-form-intro">
        A confirmation code was sent to your email ({email}).
      </div>
      <Input
        labelText="Confirmation Code"
        id="code"
        formControlProps={{
          fullWidth: true,
          value: code,
          onChange: event => setCode(event.target.value),
          className: classes.inputProps
        }}
      />
      <div className={classes.newPassword}>
        Enter your new password below:
      </div>
      <Input
        labelText="Password"
        id="password"
        formControlProps={{
          fullWidth: true,
          value: password,
          onChange: event => setPassword(event.target.value),
          className: classes.loginInputs
        }}
        inputProps={{
          type: "password",
          autoComplete: "off"
        }}
      />
      <Input
        labelText="Confirm Password"
        id="confirmPassword"
        formControlProps={{
          fullWidth: true,
          value: confirmPassword,
          onChange: event => setConfirmPassword(event.target.value),
          className: classes.loginInputs
        }}
        inputProps={{
          type: "password",
          autoComplete: "off"
        }}
      />
      <div className={classes.passwordRules}>
        Use 8 or more characters with a mix of upper and lower case letters, numbers & symbols
      </div>
      { confirmationError
        ? <div className={classes.textCenter}>
            <ErrorText>
              <p>{confirmationError}</p>
            </ErrorText>
          </div>
        : null
      }
      <Button fullWidth color="primary" type="submit" disabled={!validateResetForm()} className={classes.buttonSizes} data-cy="submit-verify">
        Verify
      </Button>
    </form>
  );
}

NewPasswordForm.propTypes = {
  email: PropTypes.string.isRequired,
  setConfirmed: PropTypes.func
};
