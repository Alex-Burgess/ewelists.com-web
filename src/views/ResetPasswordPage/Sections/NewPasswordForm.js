import React, { useState } from 'react';
import { Auth } from "aws-amplify";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import InputAdornment from "@material-ui/core/InputAdornment";
import Icon from "@material-ui/core/Icon";
// core components
import HeaderTransparent from "custom/Header/HeaderTransparent.js";
import FooterTransparent from "custom/Footer/FooterTransparent.js";
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";
import Button from "components/CustomButtons/Button.js";
import CustomInput from "components/CustomInput/CustomInput.js";

import image from "assets/img/sheep-with-shoes.jpg";

import styles from "assets/jss/custom/views/resetPasswordPageStyle.js";
const useStyles = makeStyles(styles);

export default function NewPasswordForm(props) {
  const classes = useStyles();
  const { email } = props;

  const [code, setCode] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [confirmationError, setConfirmationError] = useState('');

  const handleConfirmClick = async event => {
    event.preventDefault();

    try {
      await Auth.forgotPasswordSubmit(
        email,
        code,
        password
      );

      props.setConfirmed(true);
    } catch (e) {
      setConfirmationError(e.message);
    }
  };

  const validateResetForm = () => {
    return (
      code.length > 0 &&
      password.length > 6 &&
      password === confirmPassword
    );
  }

  return (
    <div>
      <HeaderTransparent isAuthenticated={false} />
      <div
        className={classes.pageHeader}
        style={{
          backgroundImage: "url(" + image + ")",
          backgroundSize: "cover",
          backgroundPosition: "top center"
        }}
      >
        <div className={classes.container}>
          <GridContainer justify="center">
            <GridItem xs={12} sm={12} md={5}>
              <Card className={classes.cardSignup}>
                <h2 className={classes.cardTitle}>Confirmation Code</h2>
                <CardBody>
                  <form className={classes.form} onSubmit={handleConfirmClick}>
                  <div id="confirmationMessage" className={classes.details}>
                    <p className={classes.confirmMessageSpacing}>A confirmation code was sent to your email ({email}).</p>
                  </div>
                  <CustomInput
                      id="code"
                      formControlProps={{
                        fullWidth: true,
                        className: classes.customFormControlClasses,
                        value: code,
                        onChange: event => setCode(event.target.value)
                      }}
                      inputProps={{
                        startAdornment: (
                          <InputAdornment
                            position="start"
                            className={classes.inputAdornment}
                          >
                            <Icon className={classes.inputAdornmentIcon}>
                              lock_outline
                            </Icon>
                          </InputAdornment>
                        ),
                        autoComplete: "off",
                        placeholder: "Confirmation Code..."
                      }}
                    />
                    <div className={classes.details}>
                      <p className={classes.messageSpacing}>Enter your new password below:</p>
                    </div>
                    <CustomInput
                      id="password"
                      formControlProps={{
                        fullWidth: true,
                        className: classes.customFormControlClasses,
                        value: password,
                        onChange: event => setPassword(event.target.value),
                      }}
                      inputProps={{
                        startAdornment: (
                          <InputAdornment
                            position="start"
                            className={classes.inputAdornment}
                          >
                            <Icon className={classes.inputAdornmentIcon}>
                              lock_outline
                            </Icon>
                          </InputAdornment>
                        ),
                        type: "password",
                        autoComplete: "off",
                        placeholder: "New Password..."
                      }}
                    />
                    <CustomInput
                      id="confirmPassword"
                      formControlProps={{
                        fullWidth: true,
                        className: classes.customFormControlClasses,
                        value: confirmPassword,
                        onChange: event => setConfirmPassword(event.target.value),
                      }}
                      inputProps={{
                        startAdornment: (
                          <InputAdornment
                            position="start"
                            className={classes.inputAdornment}
                          >
                            <Icon className={classes.inputAdornmentIcon}>
                              lock_outline
                            </Icon>
                          </InputAdornment>
                        ),
                        type: "password",
                        autoComplete: "off",
                        placeholder: "Confirm Password..."
                      }}
                    />
                    { confirmationError
                      ? <div id="newPasswordError" className={classes.error}>
                          <p>{confirmationError}</p>
                        </div>
                      : null
                    }
                    <div className={classes.textCenter}>
                      <Button round color="info" type="submit" disabled={!validateResetForm()}>
                        Verify
                      </Button>
                    </div>
                  </form>
                </CardBody>
              </Card>
            </GridItem>
          </GridContainer>
        </div>
        <FooterTransparent />
      </div>
    </div>
  );
}

NewPasswordForm.propTypes = {
  email: PropTypes.string.isRequired
};
