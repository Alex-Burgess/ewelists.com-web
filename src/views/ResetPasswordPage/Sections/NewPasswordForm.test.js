import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import {mount} from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Auth } from "aws-amplify";
import NewPasswordForm from './NewPasswordForm';

describe('New Password Form Snapshots', () => {
  test('Renders empty new password form', () => {
    const tree = TestRenderer
      .create(<Router><NewPasswordForm email="mock.user@gmail.com"/></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test Input Form', () => {
  const wrapper = mount(
    <Router>
      <NewPasswordForm email="mock.user@gmail.com" />
    </Router>
  );

  it('Should have 3 inputs', () => {
    expect(wrapper.find('CustomInput').length).toEqual(3);
  });

  it('Should have code input', () => {
    expect(wrapper.find('CustomInput[id="code"]').length).toEqual(1);
  });

  it('Should have password input', () => {
    expect(wrapper.find('CustomInput[id="password"]').length).toEqual(1);
  });

  it('Should have confirmPassword input', () => {
    expect(wrapper.find('CustomInput[id="confirmPassword"]').length).toEqual(1);
  });

  it('Should have 1 form submit button', () => {
    expect(wrapper.find('form').find('button').length).toEqual(1);
  });

  it('Form submit button should be disabled', () => {
    const submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);
  });

  it('Submit button should be active when all fields populated and passwords match', () => {
    wrapper.find('input[id="code"]').simulate('change', {
      target: {
        value: '123456',
      },
    });

    wrapper.update();
    let submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);

    wrapper.find('input[id="password"]').simulate('change', {
      target: {
        value: 'password',
      },
    });

    wrapper.update();
    submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);

    wrapper.find('input[id="confirmPassword"]').simulate('change', {
      target: {
        value: 'password1',
      },
    });

    wrapper.update();
    submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);

    wrapper.find('input[id="confirmPassword"]').simulate('change', {
      target: {
        value: 'password',
      },
    });

    wrapper.update();
    submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(false);
  });
});


describe('Test submit form', () => {
  const wrapper = mount(
    <Router>
      <NewPasswordForm email="mock.user@gmail.com" />
    </Router>
  );

  wrapper.find('input[id="code"]').simulate('change', {
    target: {
      value: '123456',
    },
  });

  wrapper.find('input[id="password"]').simulate('change', {
    target: {
      value: 'password',
    },
  });

  wrapper.find('input[id="confirmPassword"]').simulate('change', {
    target: {
      value: 'password',
    },
  });

  wrapper.update();

  it('should show invalid code error', () => {
    Auth.forgotPasswordSubmit = jest.fn().mockImplementation(() => {
      throw new Error("Invalid verification code provided, please try again.");
    });

    let form = wrapper.find('form');
    form.simulate('submit');

    expect(wrapper.find('#newPasswordError').text()).toEqual('Invalid verification code provided, please try again.');
  });

  it('should show invalid password error', () => {
    Auth.forgotPasswordSubmit = jest.fn().mockImplementation(() => {
      throw new Error("Password does not conform to policy: Password must have lowercase characters");
    });

    let form = wrapper.find('form');
    form.simulate('submit');

    expect(wrapper.find('#newPasswordError').text()).toEqual('Password does not conform to policy: Password must have lowercase characters');
  });
});
