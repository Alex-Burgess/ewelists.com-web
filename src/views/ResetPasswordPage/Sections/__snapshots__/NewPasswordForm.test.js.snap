// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Password Form Snapshots Renders empty new password form 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-70 makeStyles-transparent-82 makeStyles-fixed-72 MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular makeStyles-container-73 MuiToolbar-gutters"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-title-74"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="logo"
                className="makeStyles-logo-69"
                src="logo-white.png"
              />
            </a>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <div
          className="makeStyles-hidden-86 PrivateHiddenCss-smDown-168"
        >
          <div
            className="makeStyles-collapse-87"
          >
            <ul
              className="MuiList-root makeStyles-list-178 makeStyles-mlAuto-194 MuiList-padding"
            >
              <li
                className="MuiListItem-root makeStyles-listItem-179 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-181"
                  href="/listideas"
                  onClick={[Function]}
                >
                  Gift List Ideas
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-listItem-179 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-181"
                  href="/login"
                  onClick={[Function]}
                >
                  Log In
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-listItem-179 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-181"
                  href="/signup"
                  onClick={[Function]}
                >
                  Sign Up
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="makeStyles-pageHeader-2"
    style={
      Object {
        "backgroundImage": "url(sheep-with-shoes.jpg)",
        "backgroundPosition": "top center",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="makeStyles-container-1"
    >
      <div
        className="MuiGrid-root makeStyles-grid-210  MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root makeStyles-grid-314  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-5"
        >
          <div
            className="makeStyles-card-315 makeStyles-cardSignup-3"
          >
            <h2
              className="makeStyles-cardTitle-4"
            >
              Confirmation Code
            </h2>
            <div
              className="makeStyles-cardBody-330"
            >
              <form
                className="makeStyles-form-35"
                onSubmit={[Function]}
              >
                <div
                  className="makeStyles-details-45"
                  id="confirmationMessage"
                >
                  <p
                    className="makeStyles-confirmMessageSpacing-46"
                  >
                    A confirmation code was sent to your email (
                    mock.user@gmail.com
                    ).
                  </p>
                </div>
                <div
                  className="MuiFormControl-root makeStyles-formControl-345 MuiFormControl-fullWidth"
                  onChange={[Function]}
                  value=""
                >
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-338 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                    onClick={[Function]}
                  >
                    <div
                      className="MuiInputAdornment-root makeStyles-inputAdornment-33 MuiInputAdornment-positionStart"
                    >
                      <span
                        aria-hidden={true}
                        className="material-icons MuiIcon-root makeStyles-inputAdornmentIcon-34"
                      >
                        lock_outline
                      </span>
                    </div>
                    <input
                      aria-invalid={false}
                      autoComplete="off"
                      className="MuiInputBase-input MuiInput-input makeStyles-input-347 MuiInputBase-inputAdornedStart"
                      disabled={false}
                      id="code"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Confirmation Code..."
                      required={false}
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="makeStyles-details-45"
                >
                  <p
                    className="makeStyles-messageSpacing-47"
                  >
                    Enter your new password below:
                  </p>
                </div>
                <div
                  className="MuiFormControl-root makeStyles-formControl-345 MuiFormControl-fullWidth"
                  onChange={[Function]}
                  value=""
                >
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-338 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                    onClick={[Function]}
                  >
                    <div
                      className="MuiInputAdornment-root makeStyles-inputAdornment-33 MuiInputAdornment-positionStart"
                    >
                      <span
                        aria-hidden={true}
                        className="material-icons MuiIcon-root makeStyles-inputAdornmentIcon-34"
                      >
                        lock_outline
                      </span>
                    </div>
                    <input
                      aria-invalid={false}
                      autoComplete="off"
                      className="MuiInputBase-input MuiInput-input makeStyles-input-347 MuiInputBase-inputAdornedStart"
                      disabled={false}
                      id="password"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="New Password..."
                      required={false}
                      type="password"
                    />
                  </div>
                </div>
                <div
                  className="MuiFormControl-root makeStyles-formControl-345 MuiFormControl-fullWidth"
                  onChange={[Function]}
                  value=""
                >
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-338 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                    onClick={[Function]}
                  >
                    <div
                      className="MuiInputAdornment-root makeStyles-inputAdornment-33 MuiInputAdornment-positionStart"
                    >
                      <span
                        aria-hidden={true}
                        className="material-icons MuiIcon-root makeStyles-inputAdornmentIcon-34"
                      >
                        lock_outline
                      </span>
                    </div>
                    <input
                      aria-invalid={false}
                      autoComplete="off"
                      className="MuiInputBase-input MuiInput-input makeStyles-input-347 MuiInputBase-inputAdornedStart"
                      disabled={false}
                      id="confirmPassword"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Confirm Password..."
                      required={false}
                      type="password"
                    />
                  </div>
                </div>
                <div
                  className="makeStyles-textCenter-32"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-403 makeStyles-info-407 makeStyles-round-430 makeStyles-disabled-427"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="submit"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Verify
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      className="makeStyles-footer-447 makeStyles-footer-441"
    >
      <div
        className="makeStyles-container-453"
      >
         
        <div>
          <div
            className="makeStyles-left-438"
          >
            <ul
              className="MuiList-root makeStyles-list-437 MuiList-padding"
            >
              <li
                className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-435"
                  href="/"
                >
                  © 
                  2020
                   ewelists
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-435"
                  href="/contact"
                >
                  Contact Us
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-435"
                  href="/#about"
                >
                  About us
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-435"
                  href="/privacy"
                >
                  Privacy
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-436 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-435"
                  href="/terms"
                >
                  Terms
                </a>
              </li>
            </ul>
          </div>
          <div
            className="makeStyles-rightLinks-443"
          >
            <ul>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-403 makeStyles-simple-425 makeStyles-justIcon-433"
                  href="https://www.facebook.com/ewelists"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-facebook"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-403 makeStyles-simple-425 makeStyles-justIcon-433"
                  href="https://twitter.com/ewelists"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-twitter"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-403 makeStyles-simple-425 makeStyles-justIcon-433"
                  href="https://www.instagram.com/ewelists/"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-instagram"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="makeStyles-clearFix-463"
        />
      </div>
    </footer>
  </div>
</div>
`;
